### PR TITLE
1.4.11 changelog

### PR DESCRIPTION
**Pull request by alangenfeld**

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16226** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16226" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16226?utm_source=stack-comment).

File: CHANGES.md [hunk 1]
```diff
@@ -1,10 +1,54 @@
...
+
+### Experimental
+
+- New APIs for defining and executing checks on software-defined assets. These APIs are very early and subject to change. The corresponding UI has limited functionality. [Docs](<[https://docs.dagster.io](https://docs.dagster.io/_apidocs/)/_apidocs/asset-checks>)
+- Adds a new auto-materialize skip rule `AutoMaterializeRule.skip_on_not_all_parents_updated` that enforces that an asset can only be materialized if all parents have been materialized since the asset's last materialization.
+- Exposed an auto-materialize skip rule – `AutoMaterializeRule.skip_on_parent_missing` –which is already part of the behavior of the default auto-materialize policy.
+- Auto-materialize evaluation history will now be stored for 1 month, instead of 1 week.
+- The Auto-Materialize asset daemon now includes more logs about what it’s doing for each asset in each tick in the Dagster Daemon process output.
+- `MaterializeResult` has been introduced that can be returned from `@asset` / `@multi_asset` functions.
+- `AssetSpec` has been introduced as an alternative way for specifying the details of assets produced by a `@multi_asset`.
```
Comment (sryza) [inline CHANGES.md:37]: Is this part of the public API? I'm not seeing it there.

File: CHANGES.md [hunk 2]
```diff
@@ -1,10 +1,54 @@
...
+
+- [dagster-databricks] Migrated to use new official databricks Python SDK. Thanks @judahrand!
+
+### Experimental
+
+- New APIs for defining and executing checks on software-defined assets. These APIs are very early and subject to change. The corresponding UI has limited functionality. [Docs](<[https://docs.dagster.io](https://docs.dagster.io/_apidocs/)/_apidocs/asset-checks>)
+- Adds a new auto-materialize skip rule `AutoMaterializeRule.skip_on_not_all_parents_updated` that enforces that an asset can only be materialized if all parents have been materialized since the asset's last materialization.
+- Exposed an auto-materialize skip rule – `AutoMaterializeRule.skip_on_parent_missing` –which is already part of the behavior of the default auto-materialize policy.
+- Auto-materialize evaluation history will now be stored for 1 month, instead of 1 week.
+- The Auto-Materialize asset daemon now includes more logs about what it’s doing for each asset in each tick in the Dagster Daemon process output.
```
Comment (sryza) [inline CHANGES.md:35]:
```suggestion
- The auto-materialize asset daemon now includes more logs about what it’s doing for each asset in each tick in the Dagster Daemon process output.
```

**Review by sryza — APPROVED** [review]

File: CHANGES.md [hunk 3]
```diff
@@ -1,10 +1,54 @@
...
+
+- Dagster code servers now wait to shut down until any calls that they are running have finished, preventing them from stopping while in the middle of executing sensor ticks or other long-running operations.
+- The `dagster execute job` cli now accepts thanks `—op-selection` (@silent-lad!)
+- [ui] Option (Alt) + R now reloads all code locations (OSS only)
+
+### Bugfixes
+
+- Adds a check to validate partition mappings when directly constructing `AssetsDefinition` instances.
+- Assets invoked in composition functions like `@graph` and `@job` now work again, fixing a regression introduced in 1.4.5.
+
```
Comment (sryza) [inline CHANGES.md:15]: Is this newline on purpose?

File: CHANGES.md [hunk 4]
```diff
@@ -1,10 +1,54 @@
 # Changelog
 
+# 1.4.11 / 0.20.11 (libraries)
+
+### New
+
+- Dagster code servers now wait to shut down until any calls that they are running have finished, preventing them from stopping while in the middle of executing sensor ticks or other long-running operations.
+- The `dagster execute job` cli now accepts thanks `—op-selection` (@silent-lad!)
```
Comment (sryza) [inline CHANGES.md:8]: Presumably this is `--op-selection`?

**Comment by alangenfeld** [comment]
nice thanks for catching those

**Pull request merged**